api/comments/serializers.py

```diff
@@ -62,7 +62,7 @@ def get_is_abuse(self, obj):
         user = self.context['request'].user
         if user.is_anonymous():
             return False
-        return user._id in obj.reports
+        return user._id in obj.reports and not obj.reports[user._id]['retracted']
```

Comment:
replaced with gets on 65 and 190. The user._id will catch it before a `get` has a chance at a `KeyError`